### DISCUSSION TITLE
Trouble taking squid curves

### DISCUSSION THREAD

**Discussion by prakamya-agrawal** [simonsobs/sodetlib#366] [Q&A]
We're trying to take squid curves for UFM_Mv4 at SLAC and running the example written here: https://simons1.princeton.edu/docs/sodetlib/operations/squid_curves.html

This is what we get: 
![IMG_3301](https://github.com/simonsobs/sodetlib/assets/37946975/607ee125-8e12-4a96-8e26-0703c00a4d0c)
![IMG_3302](https://github.com/simonsobs/sodetlib/assets/37946975/df897d46-e487-498a-800e-c3a96885c4cb)

Any advice/guidance would be helpful!

**Comment by prakamya-agrawal** [comment]
Also when we changed it to `squid_curves.plot_squid_fit(*data, 1, S.which_on(1)[10])`, we get this error 
![IMG_3303](https://github.com/simonsobs/sodetlib/assets/37946975/aa8662a4-fcd8-406b-a0ed-43251985d173)

**Comment by jlashner** [comment]
Hi Prakamya, it seems like for some reason the analysis did not run... Do you think you can post the version or githash of sodetlib you're using? I think for now, you may need to re-run the analysis manually, which you can do like:
```python3
data = np.load(fname, allow_pickle=True).item()
squid_curves.fit_squid_curves(data, modify_data=True)
```
After that the `data` object _should_ have the `popts` key, and then hopefully you can use the squid_curve plotting functions. Can you try doing that?

**Reply by prakamya-agrawal** [reply]
sodetlib version: 0.5..0+28.glf580ca.dirty

this is what happens when we try that code
![IMG_3313](https://github.com/simonsobs/sodetlib/assets/37946975/79f90b2f-70df-4dbe-89e9-2bd09def47ec)

I think cuz it's using it's using the legacy functions in analysis? 
we're able to `import sodetlib.analysis`, but we get an error when we try to run `from sodetlib.analysis import squid_fit as sqf`

**Reply by prakamya-agrawal** [reply]
Maybe we can update sodetlib on srv-24? what would be the best way to do this?

**Reply by jlashner** [reply]
Is there any chance you can paste the error message in from the smurf-servers haha. These pics are actually pretty large files and take a long time to load on chilean wifi

**Comment by jlashner** [comment]
Ok ya so that is indeed in the most recent version of the code... I think whenever I have been running squid curves I have just been looking at the plots manually (plotting `res_freq_vs_fr` against `fluxramp_ffs`), so I either have just not run into this or have been ignoring the issue haha. If you can I'd plot these manually for now, but I will open an issue to fix the plotting functions. Do you think you can send me (in slack) the npy file with the data that you're using?

**Reply by prakamya-agrawal** [reply]
Yeah, sorry in the future, I'll paste the error directly. 
We'll plot stuff manually for now, as well! Thank you so much!

**Reply by prakamya-agrawal** [reply]
We're able to plot stuff manually, thank you!

**Comment by msilvafe** [comment]
@Prakamya01 can you please try this again and see if it fails anymore. We expect that this PR from @jlashner should have fixed the squid fit model import issues: https://github.com/simonsobs/sodetlib/commit/e804e15d9532c0f8545e92ce08dc9bb1c846f09c

If it works now we can also close the associated issue here: https://github.com/simonsobs/sodetlib/issues/367

**Reply by prakamya-agrawal** [reply]
I'm not at SLAC anymore, so I can ask Toby to run this
